zebra-chain/src/sapling/commitment.rs

```diff
@@ -120,11 +123,9 @@ impl NoteCommitment {
         // The `TryFrom<Diversifier>` impls for the `jubjub::*Point`s handles
         // calling `DiversifyHash` implicitly.
 
-        let g_d_bytes: [u8; 32] = if let Ok(g_d) = jubjub::AffinePoint::try_from(diversifier) {
-            g_d.to_bytes()
-        } else {
-            return None;
-        };
+        let g_d_bytes = jubjub::AffinePoint::try_from(diversifier)
```

Comment:
This might also need multiple retries, but I'd like to check with a cryptographer to be sure.

Reply:
It seems it will not be needed here.

The `NoteCommitment::new` takes a `Diversifier` as an argument. Creating a `Diviersifier` should be done by `Diversifier::new` where we already have the loop.

`NoteCommitment::new` is currently not used in Zebra, i guess that there is a small risk of passing an invalid `Diversifier` here but in that case i think is ok to just error here.